Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM node:20-alpine
+
+WORKDIR /usr/app
+
+COPY . .
+
+RUN ["yarn", "install", "--frozen-lockfile"]
+
+CMD [ "yarn", "start" ]
```

Comment:
Was this tested? It doesn't build the app so it will fail I think.

Reply:
By copying the .env.example (modified) and renaming it .env everything works, it has been tested. I followed Todmy's advice on the env_file rule and applied it for the sql server and the app.

Reply:
Dockerfile and docker-compose don't have anything to do with each other here.

```
# docker build .
...
 ---> e7ddd0a27a0c                                                                                                                                                                                         [0/93]
Step 2/6 : WORKDIR /usr/app
 ---> Running in 22d9bb1db54c
Removing intermediate container 22d9bb1db54c
 ---> 7284f9cc4418
Step 3/6 : COPY . .
 ---> 12fc150b0815
Step 4/6 : RUN ["yarn", "install", "--frozen-lockfile"]
 ---> Running in 86e4b40ed0fc
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "@snapshot-labs/checkpoint > express-graphql@0.12.0" has incorrect peer dependency "graphql@^14.7.0 || ^15.3.0".
warning " > ts-node@10.7.0" has unmet peer dependency "@types/node@*".
[4/4] Building fresh packages...
Done in 10.64s.
Removing intermediate container 86e4b40ed0fc
 ---> c77728e0d2c2
Step 5/6 : CMD [ "yarn", "start" ]
 ---> Running in acccbec8590c
Removing intermediate container acccbec8590c
 ---> 9432f43d1f27
Step 6/6 : EXPOSE 3000
 ---> Running in e805ee2c0130
Removing intermediate container e805ee2c0130
 ---> 10f82f8191f9
Successfully built 10f82f8191f9
➜  checkpoint-template git:(9f900f2) docker run 10f82f8191f9
yarn run v1.22.19
$ node dist/src/index.js
node:internal/modules/cjs/loader:1073
  throw err;
  ^

Error: Cannot find module '/usr/app/dist/src/index.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1070:15)
    at Module._load (node:internal/modules/cjs/loader:923:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.2.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
I agree with @Sekhmet. Each container should work fine independently, and Docker Compose is a tool used for managing the orchestration of these containers. However, if we decide to use another orchestration tool, we shouldn't necessarily have to adjust the Dockerfile. Ideally, the Dockerfile should define the environment for a standalone container, independent of the orchestration tool used.

Reply:
OK that's clearer, thanks for the explanation. Now I'd better tell you what you want and I'll take a look at how to do it.